static/src/javascripts/projects/common/modules/article/open-module.js

```diff
@@ -31,23 +33,21 @@ define([
     return {
         init: function () {
             if (config.page.openModule) {
-                spacefinder.getParaWithSpace(getSpacefinderRules()).then(function (space) {
-                    if (space) {
-                        ajax({
-                            url: config.page.openModule,
-                            crossOrigin: true,
-                            method: 'get'
-                        }).then(function (resp) {
-                            if (resp.html) {
-                                fastdom.write(function () {
-                                    $.create(resp.html)
-                                        .addClass('element--supporting')
-                                        .insertBefore(space);
-                                    $('.submeta-container--break').removeClass('submeta-container--break');
-                                });
-                            }
-                        });
-                    }
+                spaceFiller.fillSpace(getSpacefinderRules(), function (spaces) {
+                    ajax({
+                        url: config.page.openModule,
+                        crossOrigin: true,
+                        method: 'get'
+                    }).then(function (resp) {
+                        if (resp.html) {
```

Comment:
it's worth pointing out that when.js has some excellent sugarcoating on top of promises, some of which could make our code easier to read:

```
PromiseInst.then(... do some ajax stuff ...).**timeout(1000)**
PromiseInst.then(... return an array ...).**spread**(function(elem1, elem2, ...) { })
PromiseInst.**yield**(valueOrPromise)
```

as well as a "promisification" library that works on objects, functions, and callbacks.